src/message.rs

```diff
@@ -1762,7 +1766,15 @@ pub async fn markseen_msgs(context: &Context, msg_ids: Vec<MsgId>) -> Result<()>
         _curr_ephemeral_timer,
     ) in msgs
     {
-        if curr_state == MessageState::InFresh || curr_state == MessageState::InNoticed {
+        if curr_state == MessageState::InFresh
+            && matches!(
+                curr_download_state,
+                DownloadState::Available | DownloadState::InProgress
+            )
+        {
+            update_msg_state(context, id, MessageState::InNoticed).await?;
+            updated_chat_ids.insert(curr_chat_id);
+        } else if curr_state == MessageState::InFresh || curr_state == MessageState::InNoticed {
```

Comment:
With this code, `InNoticed` partially downloaded messages won't go into the first `if` (because they're not `InFresh`), so they will go into the `else if` and marked as `\Seen`, which they shouldn't.

Needs to be rewritten as sth like (pseudocode):
```
if partially_downloaded {
    // Would be nice to have a comment here, e.g.:
    // Don't mark partially downloaded messages as Seen or send a read receipt since they are not really seen by the user.
    if curr_state == InFresh {
        mark as InNoticed
    }
} else if ...
```

Reply:
Thanks, rewrote the code this way and extended the test accordingly